draft-ietf-quic-transport.md

```diff
@@ -2883,16 +2886,21 @@ needing acknowledgement are received.  The sender can use the receiver's
 Strategies and implications of the frequency of generating acknowledgments are
 discussed in more detail in {{QUIC-RECOVERY}}.
 
+An endpoint that is only sending acknowledgements will not receive
+acknowledgments from its peer unless those acknowledgements are included in
+packets with ACK-eliciting frames.  A sender SHOULD bundle ACK frames with
+other frames when possible.
+
 To limit ACK Ranges (see {{ack-ranges}}) to those that have not yet been
 received by the sender, the receiver SHOULD track which ACK frames have been
 acknowledged by its peer. The receiver SHOULD exclude already acknowledged
 packets from future ACK frames whenever these packets would unnecessarily
-contribute to the ACK frame size.
-
-Because ACK frames are not sent in response to ACK-only packets, a receiver that
-is only sending ACK frames will only receive acknowledgements for its packets if
-the sender includes them in packets with non-ACK frames.  A sender SHOULD bundle
-ACK frames with other frames when possible.
+contribute to the ACK frame size.  When the receiver is only sending
+non-ACK-eliciting packets, it can bundle a PING with a fraction of them, such
```

Comment:
By the way, the PING has an additional behavior requirement that it is ACKed ASAP, unlike other ACK-eliciting frames.  Bundling a PING forces the other endpoint to be slightly less efficient in its packet sending.  There are other alternatives/hacks like MAX_DATA(repeat latest value) that elicits ACKs but does not imply immediacy (and since it is a repeat, it does not require retransmission on loss, just like PING).

Reply:
You are correct. I remember we had a discussion about doing so, but I also do not see it landing anywhere.  So this PING as an example of what can be done sounds good.  But "MUST NOT bundle a PING" in the following sentence looks like normative language, and PING is not longer just an example there.

Maybe you want to write: "MUST NOT make every otherwise non-ACK-eliciting packet ACK-eliciting, in order to ..." or "MUST NOT add an ACK-eliciting frame to every otherwise non-ACK-eliciting packet, in order to ...".